### PR TITLE
#21 Created remaining time component

### LINKED ISSUE #21: UI Component | Display the remaining time
old hash: #17

### PR DESCRIPTION
**Pull request by SebastianBabinski1**
closes #21

### CODE REVIEW

File: styles/components/_remainingTime.scss [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  justify-content: center;
+  align-items: center;
+
+  &__counter {
+    font-size: $font-size-primary;
+    font-weight: bold;
+    color: rgba(255, 0, 0, 0.8);
+    text-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25),
+      4px 4px 40px rgba(255, 0, 0, 0.9);
+  }
```
Comment (paula0403) [inline styles/components/_remainingTime.scss:12]: Remember the order in CSS:
/* Positioning /
 / Display & Box Model /
  / Color /
  / Text /
  / Other */

File: src/app/components/RemainingTime.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+function RemainingTime() {
+  const root = document.querySelector("#swquiz-app");
+
+  const parentElement = document.createElement("div");
+  parentElement.classList.add("remaining-time");
+
+  const counter = document.createElement("div");
+  counter.textContent = "Time Left : 2 m 00 s";
+  counter.classList.add("remaining-time__counter");
+
```
Comment (paula0403) [inline src/app/components/RemainingTime.js:27]: It's a mess here. First, declaration of all variables, then methods (classList.add, textContent).

Comment (paula0403) [reply src/app/components/RemainingTime.js:27]: Maybe the timer will look better if 6s show as 06 sec (+ zero in front)?

Comment (paula0403) [reply src/app/components/RemainingTime.js:27]: (for minutes too)

**Review by paula0403 — COMMENTED** [review]
-code needs some cleanup,
-order in scss
-time display format

File: src/app/components/RemainingTime.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
+function handleTimeChange(element){
+  let timer = 120;
```
Comment (DominikNowak) [inline src/app/components/RemainingTime.js:2]: I think that here should be used `QUIZ_MAX_TIME_SECONDS` from _.env_

File: src/app/components/RemainingTime.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+  const idInterval = setInterval(() => {
+    timer--;
+
+    if(timer === 0){
+      clearInterval(idInterval);
+    }
+
+    let min = Math.floor(timer / 60);
+    let sec = timer % 60;
```
Comment (DominikNowak) [inline src/app/components/RemainingTime.js:12]: These two could be a const

**Review by DominikNowak — CHANGES_REQUESTED** [review]
Could you please fill linked issue's description with requirements? Generally looks good. RemainingTime.js looks not formatted

File: src/app/components/RemainingTime.js (hunk 2 above)
Comment (SebastianBabinski1) [reply src/app/components/RemainingTime.js:27]: > It's a mess here. First, declaration of all variables, then methods (classList.add, textContent).

In my opinion my way of formating is more transparent :)

**Review by Enessetere — CHANGES_REQUESTED** [review]
Looks good to me. Please update to develop and resolve conflicts.

**Review by paula0403 — APPROVED** [review]
It's best to put the component in an item other than swquiz-app and return it last.

File: src/app/App.js [hunk 5]
```diff
@@ -17,6 +19,8 @@ const App = ({ options }) => {
     'fa-graduation-cap',
   );
   const gameModeInfo = createGameModeName('Who is this character?');
+  
+  quiz.append(RemainingTime());
```
Comment (DominikNowak) [inline src/app/App.js:23]: There is no declared variable 'quiz' so it causes a ReferenceError

File: src/app/components/RemainingTime.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    const sec = timer % 60;
+    let minToDisplay = min;
+    let secToDisplay = sec;
+
+    if (timer === 0) {
+      clearInterval(idInterval);
+    };
+
+    if (min < 10) {
+      minToDisplay = '0' + minToDisplay;
```
Comment (DominikNowak) [inline src/app/components/RemainingTime.js:17]: Consider to use template literals

File: src/app/components/RemainingTime.js [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    };
+
+    if (sec < 10) {
+      secToDisplay = '0' + secToDisplay;
+    };
+
+    element.textContent = `Time Left: ${minToDisplay}m ${secToDisplay}s`;
```
Comment (DominikNowak) [inline src/app/components/RemainingTime.js:24]: Like here

File: src/app/components/RemainingTime.js [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+  }, 1000);
+};
+
+function RemainingTime() {
+  const parentElement = document.createElement('div');
+  const counter = document.createElement('div');
+
+  parentElement.classList.add('remaining-time');
+  counter.textContent = 'Time Left : 02 m 00 s';
+  counter.classList.add('remaining-time__counter');
```
Comment (DominikNowak) [inline src/app/components/RemainingTime.js:34]: Consider to refactor it with elementFactory

File: src/app/App.js [hunk 9]
```diff
@@ -3,8 +3,10 @@ import Logo from './components/Logo';
 import createButtonRed from './components/ButtonRed';
 import createWhiteButtonWithIcon from './components/ButtonWhiteWithIcon';
 import createGameModeName from './components/GameModeName';
+import RemainingTime from './components/RemainingTime';
```
Comment (DominikNowak) [inline src/app/App.js:6]: Consider follow naming convention as above

**Review by DominikNowak — CHANGES_REQUESTED** [review]

**Review by DominikNowak — APPROVED** [review]
Now looks ok :)

**Review by Enessetere — APPROVED** [review]
Well done. Consider changes suggested by @DominikNowak ;)

**Pull request merged**